---
title: "ADSP WES GWAS by GLMM"
author: "Xulong Wang"
date: "May 3, 2016"
output: pdf_document
---

## Score test

```{r}

library(dplyr)
library(ggplot2)
options(stringsAsFactors = F)

chrlen <- read.delim("~/Dropbox/GitHub/X/genomes/human.hg19.genome", header = F)
chrlen <- chrlen[match(paste0("chr", 1:22), chrlen$V1), ]
chrlen <- cumsum(as.numeric(chrlen$V2)) * 1e-6; names(chrlen) <- c(1:22)
chrmid <- diff(c(0, chrlen)) * 0.5 + c(0, chrlen[-length(chrlen)])

gmmat.score <- read.table("/data/xwang/wes2/glmm/gmmat.score.bed.txt", header = T)
gmmat.score <- read.table("/data/xwang/wes2/glmm/gmmat.score.bed.pc1.txt", header = T)
gmmat.score <- read.table("/data/xwang/wes2/glmm/gmmat.score.bed.pc1.pc2.txt", header = T)

gmmat.score$POS2 <- c(0, chrlen)[gmmat.score$CHR] + gmmat.score$POS * 1e-6
gmmat.score$col <- rep("o", nrow(gmmat.score))
gmmat.score$col[gmmat.score$CHR %% 2 == 1] <- "e"
gmmat.score$P <- -log10(gmmat.score$PVAL)

manhattan <- gmmat.score


png("~/Dropbox/GitHub/wes/glmm/manhattan.png", width = 2e3, height = 1e3, res = 200)

ggplot(manhattan, aes(x = POS2, y = P, color = col)) + 
  geom_point(alpha = 0.9) + geom_hline(yintercept = 7.3, color = "gold") +
  scale_x_continuous(breaks = chrmid, labels = names(chrlen)) + 
  scale_color_manual(values = c("dodgerblue3", "firebrick1")) + 
  theme_bw() + xlab("") + ylab("-log10(P)") + guides(shape = F, color = F) +
  theme(legend.key = element_blank())

dev.off()

gwas <- filter(gmmat.score, P > 3) 

group <- list(); group[[1]] <- gwas[1, ]; group_idx <- 1

for (i in 2:nrow(gwas)) {
  chromosome = gwas$CHR[i] == gwas$CHR[i-1]
  position = gwas$POS[i] - gwas$POS[i-1] < 1e6
        
  if ( all(chromosome, position) )
    group[[group_idx]] = rbind(group[[group_idx]], gwas[i, ])
  else {
    group_idx = group_idx + 1
    group[[group_idx]] = gwas[i, ]
  }
}

sapply(group, nrow)
group <- do.call(rbind, lapply(group, function(x) cbind(PEAK = x$SNP[which.max(x$P)], x)))

(peaks = as.character(unique(group$PEAK)))

load("/data/xwang/wes2/vep.rdt")
vep$UID = gsub("-.*", "", vep$Location)
vep = vep[vep$UID %in% group$SNP, ]

table = cbind(vep, group[match(vep$UID, group$SNP), ])
write.xlsx(table, file = "table(20160504)_suspicious.xlsx", sheetName = "P(1e-5)")

vep = vep[vep$UID %in% peaks, ]
unique(vep$SYMBOL)

source("~/Dropbox/GitHub/wes/gwas/loci200k.R")

lapply(peaks, function(name) {
  pdf(file = paste0(name, ".pdf"))
  make.locus.200k(name, gmmat.score)
  dev.off()
})

load("/data/xwang/wes2/igap.rdt")

igap_s1$SNP = paste(igap_s1$Chromosome, igap_s1$Position, sep = ":")
igap_s1 = igap_s1[igap_s1$SNP %in% gwas$SNP, ]

nrow(gwas)
nrow(igap_s1)

gwas_x = gwas[match(igap_s1$SNP, gwas$SNP), ]

pdf("igap.pdf")

plot(-log10(igap_s1$Pvalue), gwas_x$P, xlab = "IGAP -log10(P)", ylab = "WES -log10(P)")
abline(0, 1, col = "red")

dev.off()

```

## Genotype of extrmes

```{r}

load("./mdata.rdt")
load("./genes/cr1.rdt")

geno = z[, mdata$ADSP_SM_ID]

data = within(mdata, { geno = geno[1, ] })
data = within(mdata, { geno = geno[2, ] })

ggplot(data, aes(x = status, fill = as.factor(geno))) + geom_bar()
ggplot(data, aes(x = status, fill = as.factor(geno))) + geom_bar(position = "fill")
ggplot(data, aes(x = status, fill = as.factor(2 - geno))) + geom_bar(position = "fill")

(tbl <- table(data$geno, data$status))

```
