---
title: "Final: ADSP WES GWAS results"
author: "Xulong Wang"
date: "January 18, 2016"
output: pdf_document
---

```{r, eval = F}

setwd("/data/xwang/wes/optimizing/noK")
optimizing <- lapply(dir(), function(x) { load(x); fit })
optimizing <- lapply(optimizing, function(x) { x[, "lp"] = x[, "lp"] - min(x[, "lp"]); x})
optimizing <- as.data.frame(do.call(rbind, optimizing))
save(optimizing, file = "~/Dropbox/GitHub/wes/gwas/optimizing.rdt")

```

## Post-Pvalues

```{r}

rm(list = ls())
setwd("~/Dropbox/GitHub/wes")

load("./gwas/optimizing.rdt")

vId = rownames(optimizing)
pval = pnorm(abs(optimizing$p), sd = optimizing$se_p, lower.tail = F) * 2

lrt = 2 * (optimizing$lp + 3145.4)
pval_lrt = pchisq(lrt, df = 1, lower.tail = F) 
plot(-log10(pval), -log10(pval_lrt))

```

## Genotype of extrmes

```{r}

extreme = optimizing[pval < 5e-8, ]
rownames(extreme)

load("../data/pheno.rdt")
pheno <- pheno[match(names(e1), pheno$SRR), ]

ggplot(pheno, aes(x = status, fill = as.factor(e1))) + geom_bar()
ggplot(pheno, aes(x = status, fill = as.factor(e1))) + geom_bar(position = "fill")
(tbl <- table(pheno$e1, pheno$status))

```

## Manhattan plot

```{r}

chrlen <- read.delim("~/Dropbox/GitHub/X/genomes/human.hg19.genome", header = F)
chrlen <- chrlen[match(paste0("chr", 1:22), chrlen$V1), ]
chrlen <- cumsum(as.numeric(chrlen$V2)) * 1e-6; names(chrlen) <- c(1:22)
chrmid <- diff(c(0, chrlen)) * 0.5 + c(0, chrlen[-length(chrlen)])

chr <- as.numeric(gsub("-.*", "", vId))
pos <- as.numeric(gsub(".*-", "", vId)) * 1e-6 # Mb
pos <- c(0, chrlen)[chr] + pos

manhattan <- data.frame(uid = vId, chr = chr, pos = pos, p = optimizing$p, pval = -log10(pval))
manhattan$eff <- rep("protective", nrow(manhattan)) 
manhattan$eff[manhattan$p > 0] <- "risky"

png("./manhattan.png", width = 2e3, height = 1e3, res = 200)

ggplot(manhattan, aes(x = pos, y = pval, color = eff)) + 
  geom_point(alpha = 0.7) + geom_hline(yintercept = 7.3, color = "black") +
  scale_x_continuous(breaks = chrmid, labels = names(chrlen), limits = c(0, max(pos))) + 
  scale_color_manual(values = c("dodgerblue3", "firebrick1")) + theme_bw() + xlab("") + ylab("-log10(P)") +
  theme(legend.key = element_blank())

dev.off()

```

## Plink association test

```{r}

setwd("~/Dropbox/GitHub/wes")

options(stringsAsFactors = F)
plink <- read.table("./plink/plink.assoc.logistic", header = T)

plink <- plink[plink$TEST == "ADD", ]
plink$POS <- c(0, chrlen)[plink$CHR] + plink$BP * 1e-6

ggplot(plink, aes(x = POS, y = -log10(P))) + 
  geom_point(alpha = 0.7) + geom_hline(yintercept = 7.3, color = "red") +
  scale_x_continuous(breaks = chrmid, labels = names(chrlen))

png("./manhattan.png", width = 2e3, height = 1e3, res = 200)

ggplot(plink, aes(x = POS, y = -log10(P))) + 
  geom_point(alpha = 0.7) + geom_hline(yintercept = 7.3, color = "red") +
  scale_x_continuous(breaks = chrmid, labels = names(chrlen))

dev.off()

  
```

## SNPTEST association test

```{r}

options(stringsAsFactors = F)
snptest <- read.table("./snptest.out", header = T)
snptest$pos <- c(0, chrlen)[snptest$chromosome] + snptest$position * 1e-6

png("./manhattan.png", width = 2e3, height = 1e3, res = 200)

ggplot(snptest, aes(x = pos, y = -log10(frequentist_add_lrt_pvalue))) + 
ggplot(snptest, aes(x = pos, y = -log10(frequentist_add_wald_pvalue_1))) + 
  geom_point(alpha = 0.7) + geom_hline(yintercept = 7.3, color = "red") +
  scale_x_continuous(breaks = chrmid, labels = names(chrlen))

dev.off()
  
```

## Zoom into the APOE locus. Prior on APOE locus was too strong to drop below the threshold

```{r}

load("./Apoe/apoe.rdt")

loc_apoe <- manhattan[manhattan$uid %in% rownames(apoe), ]
qplot(x = pos, y = p_se, data = loc_apoe, geom = "point", color = eff)

```

## The top adsp variants. Top adsp variants did not have high priors, so the posterior did not improve much

```{r}

load("../Adsp/data/glmList.rdt"); adsp <- glmList$gwas

adsp <- adsp[adsp$LOD > 15, ]
adsp2 <- manhattan[match(adsp$UID, manhattan$uid), ]
adsp2 <- adsp2[!is.na(adsp2$p_se), ]

plot(adsp2$p_se, adsp2$prior, ylim = c(0, 5))
points(adsp2$p_se, adsp2$p_wo, col = "red")
abline(0, 1)

```
